docs/sagas.md

```diff
@@ -0,0 +1,239 @@
+    Init: async ({ store }) => {
+      await store.defineTable('users', {
+        indexes: { id: 'string' },
+        fields: ['mail']
+      })
+    },
+    USER_CREATED: async ({ store, executeCommand }, event) => {
+      await store.insert('users', {
+        id: event.aggregateId,
+        mail: event.payload.mail
+      })
+      await executeCommand({
+        aggregateName: 'User',
+        aggregateId: event.aggregateId,
+        type: 'requestConfirmUser',
+        payload: event.payload
+      })
+    },
+    USER_CONFIRM_REQUESTED: async ({ sideEffects, scheduleCommand }, event) => {
+      await sideEffects.sendEmail(event.payload.mail, 'Confirm mail')
+      await scheduleCommand(event.timestamp + 1000 * 60 * 60 * 24 * 7, {
+        aggregateName: 'User',
+        aggregateId: event.aggregateId,
+        type: 'forgetUser',
+        payload: {}
+      })
+    },
+    USER_FORGOTTEN: async ({ store }, event) => {
+      await store.delete('users', {
+        id: event.aggregateId
+      })
+    }
+  },
+  sideEffects: {
+    sendEmail: async (mail, content) => {
+      ...
+    }
+  }
+}
+```
+
+<!-- prettier-ignore-end -->
+
+This saga requires a new user to confirm their email address. If a user does not confirm the address in a week, the saga cancels the registration.
+
+## Define a Saga
+
+### Add a Saga to the Application
+
+You can define a saga in one of the following two ways:
+
+- In one source file as an object that contains the `handlers` and `sideEffects` objects.
+
+  ##### common/sagas/user-confirmation.saga.js:
+
+  ```js
+  export default {
+    handlers: {
+      // Event handlers implementation
+    }
+    sideEffects: {
+      // Side effects implementation
+    }
+  }
+  ```
+
+- In two separate files.
+
+  ##### common/sagas/user-confirmation.handlers.js:
+
+  ```js
+  export default {
+    // Event handlers implementation
+  }
+  ```
+
+  ##### common/sagas/user-confirmation.side-effects.js:
+
+  ```js
+  export default {
+    // Side effects implementation
+  }
+  ```
+
+### Initialize the Storage
+
+Every saga should define an `Init` function that initializes the saga's persistent storage:
+
+[mdis]: # '../tests/saga-sample/saga.js#init'
+
+```js
+Init: async ({ store }) => {
+  await store.defineTable('users', {
+    indexes: { id: 'string' },
+    fields: ['mail']
+  })
+},
+```
+
+### Handle Events
+
+An event handler function runs for every occurrence of a specific event. It has the following general structure:
+
+```js
+EVENT_NAME: async (
+  { executeCommand, scheduleCommand, store, sideEffects },
+  event
+) => {
+  // Event handler logic
+}
+```
+
+As a first argument, an event handler receives an object that provides access to the saga-related API. This API includes the following functions:
+
+- `executeCommand` - Sends a command with the specified payload to an aggregate.
+- `scheduleCommand` - Similar to `executeCommand` but delays the command execution until the specified moment in time.
```

Comment:
Change "Similar to `executeCommand` but" to "Similar to `executeCommand`, but"
(I inserted a comma)
Change "delays the command execution until the specified moment in time" to "delays command execution until a specified moment in time".

Reply:
done